### PR TITLE
Blog post with guidelines on SaaS REST API design

### PR DESCRIPTION
**Pull request by bittrance**
This blog post provides guidelines for RESTful APIs for software-as-a-service offerings, with the goal to maximize their life span and support rapid evolution throughout it. The idea is to be a bit more concrete than other guidelines, primarily by being more focused.

### CODE REVIEW

File: blog/2022-04-25designing-restful-apis-for-cloud-services.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+### The caller is responsible for the use case
+
+_The RESTful API concerns itself with effective access to the data model (the "resources"). Generally speaking, the use case is the caller's concern. For example, the caller may be required to combine data from numerous API calls, read more data than it needs and perform its own sorting._
+
+**Motivation**: An endpoint that is optimized for one use case will be hard pressed to accommodate a second use case. It will be hard to avoid adding a second similar endpoint, risking divergence. Furthermore, use cases will evolve over time and if too much of processing, filtering and sorting quirks is handled by the endpoint, it is very easy to end up in a situation where we have to spend our time optimizing specific database queries for individual use cases rather than improving the performance for all callers by refactoring storage.
+
+Properly implemented RESTful APIs have a good chance of ageing gracefully. By pushing parts of the business logic to the caller (e.g. a batch job or a [backend-for-frontend](https://samnewman.io/patterns/architectural/bff/) it ensures that the RESTful API can be reused across many use cases.
```
Comment (simongottschlag) [inline blog/2022-04-25designing-restful-apis-for-cloud-services.md:56]: Missing end paranthesis

**Review by simongottschlag — APPROVED** [review]
I love it! Good job 👍

File: blog/2022-04-25designing-restful-apis-for-cloud-services.md (hunk 1 above)
Comment (bittrance) [reply blog/2022-04-25designing-restful-apis-for-cloud-services.md]: Fixed.

**Review by simongottschlag — APPROVED** [review]
I love it 😊❤️

**Pull request merged**